pkg/operator/ceph/object/spec.go

```diff
@@ -318,37 +318,77 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
-					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]),
+				cephconfig.NewFlag("rgw crypt s3 kms backend",
+					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.Provider]),
+				cephconfig.NewFlag("rgw crypt vault addr",
+					c.store.Spec.Security.KeyManagementService.ConnectionDetails[api.EnvVaultAddress]),
 			)
 		}
-		if c.store.Spec.Security.KeyManagementService.IsTLSEnabled() &&
-			c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 16, Minor: 2, Extra: 6}) {
+		if len(c.store.Spec.SSEType) != 0 && strings.Contains(c.store.Spec.SSEType, "s3") {
 			container.Args = append(container.Args,
-				cephconfig.NewFlag("rgw crypt vault verify ssl", "true"))
-			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultClientCert) != "" {
+				cephconfig.NewFlag("rgw crypt sse s3 backend",
+					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.Provider]),
+				cephconfig.NewFlag("rgw crypt sse s3 vault addr",
+					c.store.Spec.Security.KeyManagementService.ConnectionDetails[api.EnvVaultAddress]),
+			)
+		}
+		if c.store.Spec.Security.KeyManagementService.IsTokenAuthEnabled() {
+			if len(c.store.Spec.SSEType) == 0 || strings.Contains(c.store.Spec.SSEType, "kms") {
 				container.Args = append(container.Args,
-					cephconfig.NewFlag("rgw crypt vault ssl clientcert", path.Join(rgwVaultDirName, kms.VaultCertFileName)))
+					cephconfig.NewFlag("rgw crypt vault auth", kms.KMSTokenSecretNameKey),
+					cephconfig.NewFlag("rgw crypt vault token file",
+						path.Join(rgwVaultDirName, kms.VaultFileName)),
+					cephconfig.NewFlag("rgw crypt vault prefix", c.vaultPrefixRGW()),
+					cephconfig.NewFlag("rgw crypt vault secret engine",
+						c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]),
+				)
 			}
-			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultClientKey) != "" {
+			if len(c.store.Spec.SSEType) != 0 && strings.Contains(c.store.Spec.SSEType, "s3") {
 				container.Args = append(container.Args,
-					cephconfig.NewFlag("rgw crypt vault ssl clientkey", path.Join(rgwVaultDirName, kms.VaultKeyFileName)))
+					cephconfig.NewFlag("rgw crypt sse s3 vault auth", kms.KMSTokenSecretNameKey),
+					cephconfig.NewFlag("rgw crypt sse s3 vault token file",
+						path.Join(rgwVaultDirName, kms.VaultFileName)),
+					cephconfig.NewFlag("rgw crypt sse s3 vault prefix", c.vaultPrefixRGW()),
+					cephconfig.NewFlag("rgw crypt sse s3 vault secret engine",
+						c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]),
+				)
 			}
-			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultCACert) != "" {
+		}
+		if c.store.Spec.Security.KeyManagementService.IsTLSEnabled() &&
+			c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 16, Minor: 2, Extra: 6}) {
```

Comment:
Use a named variable for `cephver.CephVersion{Major: 16, Minor: 2, Extra: 6}`, like `cephVersionMinSSE`?